services/app-api/handlers/reports/fetch.ts

```diff
@@ -2,19 +2,33 @@ import handler from "../handler-lib";
 import dynamoDb from "../../utils/dynamo/dynamodb-lib";
 import s3Lib from "../../utils/s3/s3-lib";
 import { AnyObject, S3Get, StatusCodes } from "../../utils/types/types";
-import { error, buckets } from "../../utils/constants/constants";
+import {
+  error,
+  buckets,
+  reportBuckets,
+  reportTables,
+} from "../../utils/constants/constants";
 
 export const fetchReport = handler(async (event, _context) => {
   let status, body;
-  if (!event?.pathParameters?.state! || !event?.pathParameters?.id!) {
+  if (
+    !event?.pathParameters?.reportType! ||
+    !event?.pathParameters?.state! ||
+    !event?.pathParameters?.id!
+  ) {
```

Comment:
Since codeclimate is being mad about this and now that theres a few reasons to throw an error, maybe we pull this out to a function or something so this can read like "If (!validReport) throw new Error(error.NO_KEY)"

Reply:
Good call! I added a method under `validation.ts` for this.